homeassistant/components/matter/fan.py

```diff
@@ -0,0 +1,306 @@
+        """Oscillate the fan."""
+        await self.matter_client.write_attribute(
+            node_id=self._endpoint.node.node_id,
+            attribute_path=create_attribute_path_from_attribute(
+                self._endpoint.endpoint_id,
+                clusters.FanControl.Attributes.RockSetting,
+            ),
+            value=self.get_matter_attribute_value(
+                clusters.FanControl.Attributes.RockSupport
+            )
+            if oscillating
+            else 0,
+        )
+
+    async def async_set_direction(self, direction: str) -> None:
+        """Set the direction of the fan."""
+        await self.matter_client.write_attribute(
+            node_id=self._endpoint.node.node_id,
+            attribute_path=create_attribute_path_from_attribute(
+                self._endpoint.endpoint_id,
+                clusters.FanControl.Attributes.AirflowDirection,
+            ),
+            value=clusters.FanControl.Enums.AirflowDirectionEnum.kReverse
+            if direction == DIRECTION_REVERSE
+            else clusters.FanControl.Enums.AirflowDirectionEnum.kForward,
+        )
+
+    async def _set_wind_mode(self, wind_mode: str | None) -> None:
+        """Set wind mode."""
+        if wind_mode == PRESET_NATURAL_WIND:
+            wind_setting = WindBitmap.kNaturalWind
+        elif wind_mode == PRESET_SLEEP_WIND:
+            wind_setting = WindBitmap.kSleepWind
+        else:
+            wind_setting = 0
+        await self.matter_client.write_attribute(
+            node_id=self._endpoint.node.node_id,
+            attribute_path=create_attribute_path_from_attribute(
+                self._endpoint.endpoint_id,
+                clusters.FanControl.Attributes.WindSetting,
+            ),
+            value=wind_setting,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        if not hasattr(self, "_attr_preset_modes"):
+            self._calculate_features()
+        if self._attr_supported_features & FanEntityFeature.DIRECTION:
+            direction_value = self.get_matter_attribute_value(
+                clusters.FanControl.Attributes.AirflowDirection
+            )
+            self._attr_current_direction = (
+                DIRECTION_REVERSE
+                if direction_value
+                == clusters.FanControl.Enums.AirflowDirectionEnum.kReverse
+                else DIRECTION_FORWARD
+            )
+        if self._attr_supported_features & FanEntityFeature.OSCILLATE:
+            self._attr_oscillating = (
+                self.get_matter_attribute_value(
+                    clusters.FanControl.Attributes.RockSetting
+                )
+                != 0
+            )
+
+        # speed percentage is always provided
+        current_percent = self.get_matter_attribute_value(
+            clusters.FanControl.Attributes.PercentCurrent
+        )
+        self._attr_percentage = None if current_percent == 255 else current_percent
```

Comment:
I didn't really see 255 in the spec, is that something you encountered in testing? If so, maybe worth a comment here what's up with this value exactly.

Reply:
Yeah, I get back 255 if the speed is controlled by auto mode

Reply:
added a comment